multi-skill/src/bin/codechef_checkpoints.rs

```diff
@@ -1,61 +1,42 @@
 use multi_skill::data_processing::{
     get_dataset_by_name, read_csv, try_write_slice_to_file, write_json,
 };
-use multi_skill::metrics::{compute_metrics_custom, PerformanceReport};
 use multi_skill::summary::make_leaderboard;
-use multi_skill::systems::{get_rating_system_by_name, simulate_contest, Player, PlayersByName};
-
-use serde::{Deserialize, Serialize};
-use std::cell::RefCell;
-
-#[derive(Serialize, Deserialize, Clone, Debug)]
-struct SimplePlayer {
-    handle: String,
-    cur_mu: f64,
-    cur_sigma: f64,
-}
-
-fn make_checkpoint(players: Vec<SimplePlayer>) -> PlayersByName {
-    players
-        .into_iter()
-        .map(|simp| {
-            let player = Player::with_rating(simp.cur_mu, simp.cur_sigma, 0);
-            (simp.handle, RefCell::new(player))
-        })
-        .collect()
-}
+use multi_skill::systems::{simulate_contest, EloMMR, EloMMRVariant};
 
 fn main() {
     tracing_subscriber::fmt::init();
 
-    // Parse arguments, prepare rating system and datasets
-    let args: Vec<String> = std::env::args().collect();
-    if args.len() != 2 {
-        tracing::error!("Usage: {} system_name", args[0]);
-        return;
-    }
-    let system = &args[1];
-    let system = get_rating_system_by_name(system).unwrap();
     let dataset = get_dataset_by_name("codechef").unwrap();
     let mut mu_noob = 1500.;
-    let sig_noob = 350.;
+    let sig_noob = 325.;
+    let weight_limit = 0.2;
+    let sig_limit = 75.;
+    let system = EloMMR {
+        weight_limit,
+        sig_limit,
+        drift_per_sec: 0.,
+        split_ties: false,
+        subsample_size: 512,
+        subsample_bucket: 0.5,
+        variant: EloMMRVariant::Logistic(0.1),
+    };
+
     let mut players = std::collections::HashMap::new();
-    let mut avg_perf = compute_metrics_custom(&mut players, &[]);
 
     // Get list of contest names to compare with Codechef's rating system
     let paths = std::fs::read_dir("/home/work_space/elommr-data/ratings").unwrap();
     let mut checkpoints = std::collections::HashSet::<String>::new();
     for path in paths {
         if let Some(contest_name) = path.unwrap().path().file_stem() {
```

Comment:
Is the intention to abort on all failures? If so, we can do
```
let contest_name = path.unwrap().path().file_stem().unwrap().to_str().unwrap();
checkpoints.insert(contest_name.to_owned());
```
And just to avoid the `path.path()`, maybe `path` could be named to something like `file_entry`.